src/fstab-generator/fstab-generator.c

```diff
@@ -936,6 +936,10 @@ static int parse_fstab_one(
                         mount_is_network(fstype, options) ? SPECIAL_REMOTE_FS_TARGET :
                                                             SPECIAL_LOCAL_FS_TARGET;
 
+        /* Setting nofail for /sysroot/usr doesn't make sense so let's just silently ignore it. */
+        if (is_sysroot_usr && FLAGS_SET(flags, MOUNT_NOFAIL))
+                flags &= ~MOUNT_NOFAIL;
```

Comment:
Would it be possible to print a warning so that the admin gets aware that nofail doesn't make sense?